lib/declare_schema/model/field_spec.rb

```diff
@@ -60,18 +60,22 @@ def initialize(model, name, type, position: 0, **options)
         @position = position
         @options = options.dup
 
-        @options.has_key?(:null) or @options[:null] = false
+        @options.has_key?(:null) or @options[:null] = Generators::DeclareSchema::Migration::Migrator.default_null
+        @options[:null].nil? and raise "null: must be provided for field #{model}##{@name}: #{@options.inspect} since Generators::DeclareSchema::Migration::Migrator#default_null is set to 'nil'; do you want `null: false`?"
```

Comment:
Oh, I just noticed that the config is in the `Generators` namespace. I don't think it should be. (I hope to fix that someday and not use that namespace at all.)
Since it's in the gem ::DeclareSchema, the config should live there too:
```
DeclareSchema.default_null = false
etc
```

Reply:
^ Does the above sound reasonable? I think it's surprising for the `DeclareSchema` gem to have any code or variables outside of that namespace. I hope we can remove the `Generators` reference entirely at some point I tried once but timed out after 20 minutes.

Reply:
Agreed, I think everything should live within the `DeclareSchema` namespace.

Reply:
It looks as though the `Generators` namespace was reflecting the `ActiveRecord::Generators::Migration` and `ActiveRecord::Generators::Model` structure. But even so the pattern should follow as `DeclareSchema::Generators::Migration` not `Generators::DeclareSchema::Migration`.

Reply:
When I tried moving from `Generators::DeclareSchema` to `DeclareSchema::Generators`, I got hung up because Rails complained; it was expecting the former. I didn't have time to track down if there's a way to tune Rails to not expect that. But in any case, let's go with this:
1. [this PR] Move the tunable settings (part of the external interface) into `::DeclareSchema`.
2. [later] Move everything into the `::DeclareSchema` namespace, with the minimal hook for Rails to find it. The cool thing in Ruby is that you can alias one constant to another with simple assignment, so all it will take is this:
```
module Generators
  module DeclareSchema
    Migration = :: DeclareSchema::Generators::Migration
  end
end
```

Reply:
I went ahead and moved the tunable setting for [default_text_limit](https://github.com/Invoca/declare_schema/pull/61/commits/78d103440c452ed9dd0976ddc531274eaa9fa563) in this commit. The same pattern should follow for the remaining settings.

Reply:
I'm also wondering whether or not there is a need to deprecate the `character set` and `collation` settings since they already exist in previous release version tunable in the `Generators::DeclareSchema::Migration::Migrator` namespace

Reply:
Ideally, yes. Can you make those methods that delegate to the new location? I think that's just:
```
module Generators
  module DeclareSchema
    def default_character_set=(character_set)
      ::DeclareSchema.default_character_set = character_set
    end
    deprecate :default_character_set=, ...
  end
end
```